Description: Add support for LXQt as a Linux desktop environment
 This change makes LXQt use the Breeze icon theme, and adds it as a
 desktop environment along the other supported DEs.
Author: Simon Quigley <tsimonq2@ubuntu.com>
Origin: backport
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1799001
Applied-Upstream: commit:ac9c14d, commit:b63d48a
Last-Update: 2018-10-20
--- a/vcl/inc/unx/desktops.hxx
+++ b/vcl/inc/unx/desktops.hxx
@@ -34,7 +34,8 @@ enum SAL_DLLPUBLIC_RTTI DesktopType {
     DESKTOP_TDE,
     DESKTOP_KDE3,
     DESKTOP_KDE4,
-    DESKTOP_KDE5
+    DESKTOP_KDE5,
+    DESKTOP_LXQT
 }; // keep in sync with desktop_strings[] in salplug.cxx
 
 #endif // INCLUDED_VCL_INC_UNX_DESKTOPS_HXX
--- a/vcl/source/app/IconThemeSelector.cxx
+++ b/vcl/source/app/IconThemeSelector.cxx
@@ -57,7 +57,8 @@ IconThemeSelector::GetIconThemeForDeskto
     (void)desktopEnvironment;
 #else
     if ( desktopEnvironment.equalsIgnoreAsciiCase("kde4") ||
-         desktopEnvironment.equalsIgnoreAsciiCase("kde5") ) {
+         desktopEnvironment.equalsIgnoreAsciiCase("kde5") ||
+         desktopEnvironment.equalsIgnoreAsciiCase("lxqt") ) {
         r = "breeze";
     } else
     if ( desktopEnvironment.equalsIgnoreAsciiCase("macosx") ) {
--- a/vcl/unx/generic/plugadapt/salplug.cxx
+++ b/vcl/unx/generic/plugadapt/salplug.cxx
@@ -196,7 +196,9 @@ static SalInstance* autodetect_plugin()
               desktop == DESKTOP_TDE ||
               desktop == DESKTOP_MATE )
         pList = pStandardFallbackList;
-    else if( desktop == DESKTOP_KDE4 || desktop == DESKTOP_KDE5 )
+    else if( desktop == DESKTOP_KDE4 ||
+              desktop == DESKTOP_KDE5 ||
+              desktop == DESKTOP_LXQT )
         pList = pKDEFallbackList;
 
     SalInstance* pInst = nullptr;
@@ -293,7 +295,8 @@ const OUString& SalGetDesktopEnvironment
     static const char * const desktop_strings[] = {
         "none", "unknown", "GNOME", "UNITY",
         "XFCE", "MATE", "TDE",
-        "KDE", "KDE4", "KDE5" };
+        "KDE", "KDE4", "KDE5",
+        "LXQT" };
     static OUString aRet;
     if( aRet.isEmpty())
     {
--- a/vcl/unx/generic/desktopdetect/desktopdetector.cxx
+++ b/vcl/unx/generic/desktopdetect/desktopdetector.cxx
@@ -289,6 +289,8 @@ DESKTOP_DETECTOR_PUBLIC DesktopType get_
     {
         OString aOver( pOverride );
 
+        if ( aOver.equalsIgnoreAsciiCase( "lxqt" ) )
+            return DESKTOP_LXQT;
         if ( aOver.equalsIgnoreAsciiCase( "tde" ) )
             return DESKTOP_TDE;
         if ( aOver.equalsIgnoreAsciiCase( "kde5" ) )
@@ -336,8 +338,10 @@ DESKTOP_DETECTOR_PUBLIC DesktopType get_
         {
             if ( rCurrentDesktopStr.equalsIgnoreAsciiCase( "unity" ) )
                 return DESKTOP_UNITY;
-            else if ( rCurrentDesktopStr.equalsIgnoreAsciiCase( "gnome") )
+            else if ( rCurrentDesktopStr.equalsIgnoreAsciiCase( "gnome" ) )
                 return DESKTOP_GNOME;
+            else if ( rCurrentDesktopStr.equalsIgnoreAsciiCase( "lxqt" ) )
+                return DESKTOP_LXQT;
         }
     }
 
@@ -350,6 +354,8 @@ DESKTOP_DETECTOR_PUBLIC DesktopType get_
         ret = DESKTOP_MATE;
     else if ( aDesktopSession.equalsIgnoreAsciiCase( "xfce" ) )
         ret = DESKTOP_XFCE;
+    else if ( aDesktopSession.equalsIgnoreAsciiCase( "lxqt" ) )
+        ret = DESKTOP_LXQT;
     else
     {
         // these guys can be slower, with X property fetches,
